presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -93,6 +102,15 @@
     @GuardedBy("this")
     private SetMultimap<ConnectorId, InternalNode> activeNodesByConnectorId;
 
+    @GuardedBy("this")
+    private SetMultimap<ConnectorId, InternalNode> nodesByConnectorId;
+
+    @GuardedBy("this")
+    private SetMultimap<String, ConnectorId> connectorIdsByNodeId;
+
+    @GuardedBy("this")
+    private Map<String, InternalNode> nodes;
```

Comment:
If it is a set, then the nodes are not replaced when the hostport of the node changes. In a set, we don't replace an entry if it's already existing. Lets say initially Node1 is on hostA. But now if it is on hostB, then we need to update the InternalNode entry in the set. This will not happen if it is a set. If new entry is equal to existing entry, then we will not replace in a set.